src/databricks/labs/ucx/source_code/site_packages.py

```diff
@@ -15,6 +18,51 @@
 )
 
 
+class PipResolver(BaseDependencyResolver):
+    # TODO: use DistInfoResolver to load wheel/egg/pypi dependencies
+    # TODO: https://github.com/databrickslabs/ucx/issues/1642
+    # TODO: https://github.com/databrickslabs/ucx/issues/1643
+    # TODO: https://github.com/databrickslabs/ucx/issues/1640
+
+    def __init__(
+        self,
+        file_loader: FileLoader,
+        next_resolver: BaseDependencyResolver | None = None,
+    ):
+        super().__init__(next_resolver)
+        self._file_loader = file_loader
+
+    def with_next_resolver(self, resolver: BaseDependencyResolver) -> BaseDependencyResolver:
+        return PipResolver(self._file_loader, resolver)
+
+    def resolve_library(self, path_lookup: PathLookup, name: str) -> MaybeDependency:
+        path_lookup.append_path(self._temporary_virtual_environment)
```

Comment:
Should the resolve library append the path? Or should it be done outside, i.e. first resolve library then append the path if the library can actually be resolved